core/include/vecmem/containers/impl/static_array.ipp

```diff
@@ -0,0 +1,237 @@
+/**
+ * VecMem project, part of the ACTS project (R&D line)
+ *
+ * (c) 2021 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+
+#pragma once
+
+#include "vecmem/utils/types.hpp"
+
+#include <cstddef>
+#include <type_traits>
+#include <stdexcept>
+
+namespace vecmem {
+    template<typename T, std::size_t N>
+    VECMEM_HOST_AND_DEVICE
+    static_array<T, N>::static_array(
+        void
+    ) {
+        /*
+         * This function does quite literally nothing, and leaves the array's
+         * contents uninitialized.
+         */
+    }
```

Comment:
I think we'll need to replace it with a declaration that the default constructor should be `= default`. Otherwise SYCL will not accept such objects as "input variables" to SYCL kernels.

But we can sort that out once we get there...